collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2343,53 +2333,50 @@ public void denied() {
                         }
                     });
                 } else {
-                    Intent reqIntent = getIntent();
-                    boolean showFirst = reqIntent.getBooleanExtra("start", false);
-
-                    if (!showFirst) {
-                        // we've just loaded a saved form, so start in the hierarchy view
-                        String formMode = reqIntent.getStringExtra(ApplicationConstants.BundleKeys.FORM_MODE);
-                        if (formMode == null || ApplicationConstants.FormModes.EDIT_SAVED.equalsIgnoreCase(formMode)) {
-                            formControllerAvailable(formController);
-
-                            identityPromptViewModel.requiresIdentityToContinue().observe(this, requiresIdentity -> {
-                                if (!requiresIdentity) {
-                                    if (!allowMovingBackwards) {
-                                        // we aren't allowed to jump around the form so attempt to
-                                        // go directly to the question we were on last time the
-                                        // form was saved.
-                                        // TODO: revisit the fallback. If for some reason the index
-                                        // wasn't saved, we can now jump around which doesn't seem right.
-                                        FormIndex formIndex = SaveFormIndexTask.loadFormIndexFromFile();
-                                        if (formIndex != null) {
-                                            formController.jumpToIndex(formIndex);
-                                            onScreenRefresh();
-                                            return;
-                                        }
-                                    }
-
-                                    formController.getAuditEventLogger().logEvent(AuditEvent.AuditEventType.FORM_RESUME, true, System.currentTimeMillis());
-                                    formController.getAuditEventLogger().logEvent(AuditEvent.AuditEventType.HIERARCHY, true, System.currentTimeMillis());
-                                    startActivityForResult(new Intent(this, FormHierarchyActivity.class), RequestCodes.HIERARCHY_ACTIVITY);
-                                }
-                            });
+                    // Returning to the app after process death
+                    if (formController.getFormIndex().isInForm()) {
```

Comment:
If we are here (just after loading a form) and our index is in form (that mean it was saved and now restored) that means it was caused by process death. I hope there are no other cases :)

Reply:
Could you add more details to the comment here? I'm not sure I follow exactly how we end up with an index in the form in this case but not if I kill the app and then open the form again? 

It'd also good to point out here that in the activity death scenario we don't need to reload the form so we don't end up here.

Reply:
It's not relevant anymore.